### DISCUSSION TITLE
Problem using RMxprt User Defined Primitive in Maxwell 3D

### DISCUSSION THREAD

**Discussion by Porta-L** [ansys/pyaedt#3941] [Q&A] [answered]
This describes a problem encountered when creating a coil in Maxwell 3D using the LapCoil User Defined Primitive from RMxprt.
When creating a LapCoil object, the InfoCoil parameter is used to define the type of object to be created according to:

- winding => InfoCoil = 0
- coil => InfoCoil = 1
- terminal => InfoCoil = 2
- etc.

To create a single coil, I set the _InfoCoil_ parameter to _1_ in the _udp_parameters_list_ of the following function:
_Modeler3D.create_udp(udp_dll_name, udp_parameters_list, upd_library='syslib', name=None)_

The object is then created and visible in the GUI, but appears as a complete winding and not a coil. When you open the parameter list from the GUI, you can see that the InfoCoil parameter has not been correctly transmitted from PyAEDT and does not have the correct value.

You can find a simple example in the attached script where the goal is to create a coil and a terminal.
[test_udp.zip](https://github.com/ansys/pyaedt/files/13547642/test_udp.zip)

Here are more information about the function used:
[Ansys Maxwell Doc](https://ansyshelp.ansys.com/account/secured?returnurl=/Views/Secured/Electronics/v232/en/home.htm%23../Subsystems/Maxwell/Content/UDPsforSlotCoresandCoils.html)
[PyAEDT Doc](https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.modeler.modeler3d.Modeler3D.create_udp.html)

Any support is welcome, thank you in advance.

**Comment by DaveTwyman** [comment]
I've taken a look and can't immediately see what the issue is.
I've created Issue (https://github.com/ansys/pyaedt/issues/3962), let's see if its something simple we've both missed

**Comment by DaveTwyman** [comment, marked as answer]
Hello @Porta-L ,
This looks to be a bug in the area of AEDT UDP handling . We will raise this internally and get it fixed.
Until then you can adjust the value of 'InfoCoil' using the second line below

coil = app.modeler.create_udp(udp_dll_name="RMxprt/LapCoil", udp_parameters_list=parameters_list, upd_library='syslib', name='coil')
app.modeler[coil].history().props['InfoCoil'] = '2'

**Reply by DaveTwyman** [reply]
I'll mark this discussion as closed, the issue has a workaround and the bug associated will still persist until resolved.
Of course just come through again if you spot anything else or there are follow on issues.
